lib/db.js

```diff
@@ -349,45 +358,36 @@ class LimitDBRedis extends EventEmitter {
     }
 
     const count = params.count || 50;
-    const transform = this.buildStatusResponse(bucket, prefixLength);
+    const type = params.type;
 
-    const nodes = this.redis.nodes ? this.redis.nodes('slave') : [this.redis];
-    const errors = [];
-    async.map(nodes, (node, cb) => {
-      if (!node.bucketStatus) {
-        node.defineCommand('bucketStatus', {
-          numberOfKeys: 0,
-          lua: STATUS_LUA
-        });
-      }
-      node.bucketStatus(prefix, count, (err, results) => {
-        if (err) {
-          errors.push(err);
-          this.emit('error', err);
-        }
-        return cb(null, results || []);
-      });
-    }, (err, results) => {
-      if (err) {
-        // Should not happen
-        return callback(err);
-      }
+    const rStream = this.scanStream({ prefix });
+    const flattenStream = new streamUtils.BreakReadBatchStream();
+    const keyInfoStream = new streamUtils.StatusKeyDetails({ redis: this.redis, type, prefixLength});
 
-      const items = [];
-      for (let i = 0; i < results.length; i++) {
-        for (let j = 0; j < results[i].length; j++) {
-          const match = results[i][j];
-          items.push(transform(match[2], parseInt(match[0], 10), parseInt(match[1], 10)));
-          if (items.length === count) {
-            break;
-          }
+    const items = [];
+    let errors = [];
+
+    rStream.on('error', (err) => {
+      errors.push(err);
+    });
+    flattenStream.on('error', (err) => {
+      errors.push(err);
+    });
+    keyInfoStream.on('error', (err) => {
+      errors.push(err);
+    })
+
+    rStream.pipe(flattenStream).pipe(keyInfoStream)
+      .on('data', (data) => {
+        if (items.length < count) {
+          items.push(this.buildStatusResponse(bucket, data));
         }
-        if (items.length === count) {
-          break;
+        if (items.length >= count) {
+          rStream.destroy();
```

Comment:
This is not available in node 6. It was added in node 8. Are you sure it is fine to use it?

Reply:
We might want too reduce the count and add a test case for this if it is not already there.

Reply:
Solid catch. No bueno.

Reply:
It's already there the test. But `destroy` not being available in 6 makes it... :(